lib/checkcondition.cpp

```diff
@@ -1188,12 +1187,13 @@ void CheckCondition::alwaysTrueFalse()
                 continue;
             if (Token::Match(tok, "! %num%|%bool%"))
                 continue;
+            if (Token::Match(tok, "%oror%|&&") && !Token::Match(tok->astOperand1(), "%char%") && !Token::Match(tok->astOperand2(), "%char%"))
```

Comment:
Why do you check if a operand is a '%char%`? Why not just:

    if (Token::Match(tok, "%oror%|&&"))
        continue;

Reply:
So it passes [this test case](https://github.com/danmar/cppcheck/blob/master/test/testcondition.cpp#L2431) with `if(1 && 'c'){}`.

Reply:
ok. You can remove that warning message. It seems redundant to write 2 warnings about that condition. I read https://trac.cppcheck.net/ticket/7750 and I am not sure this checking belongs in alwaysTrueFalse. It seems it would be much better to write something like `Conversion of char literal 'c' to bool always evaluates to true.`

Reply:
So we will add a bailout for char literals in `alwaysTrueFalse` either now or later. And make sure the problem is properly reported.

Reply:
I moved these warnings away from `alwaysTrueFalse` with 77b653bf94def8a49773380c31932dda00fc4862. Sorry for the trouble.